gammapy/cube/fit.py

```diff
@@ -867,11 +868,20 @@ def to_image(self, spectrum=None, keepdims=True):
         dataset : `MapDataset`
             Map dataset containing images.
         """
+        self.counts.data[~self.mask_safe.data] = 0
+        self.background_model.evaluate().data[~self.mask_safe.data] = 0
+
         counts = self.counts.sum_over_axes(keepdims=keepdims)
         exposure = _map_spectrum_weight(self.exposure, spectrum)
         exposure = exposure.sum_over_axes(keepdims=keepdims)
         background = self.background_model.evaluate().sum_over_axes(keepdims=keepdims)
 
+        slices_num = self.mask_safe.geom.get_axis_by_name("ENERGY").nbin
```

Comment:
Note that the `+` operator corresponds to the logical or operation:
```python3
import numpy as np
mask_1 = np.array([[1, 0]], dtype=bool)
mask_2 = np.array([[1], [0]], dtype=bool)

print(mask_1 + mask_2)
print()
print(np.logical_or(mask_1, mask_2))
```

So you can basically use `.sum_over_axis()` for the mask as well (as long as the `dtype` is `bool`)